### PR TITLE
pass variables overrides via environment variables

### PR DESCRIPTION
**Pull request by HugoPct**
### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/synthetics/private_locations/configuration.md [hunk 1]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
...
+
+`variableOverride`
+: **Type**: String <br>
+Overrides the variables used in tests running on the Private Location. Format: `SECRET_VAR=value`.
+All variables imported this way will be obfuscated.
+
+`environmentVariableOverride`
+: **Type** String <br>
+Allows the usage of environment variables in the Private Location. It requires the environment variables to be imported in the containerized environment.
+Example for Docker: `docker run --env SECRET=value --environmentVariableOverride synthetics-private-location-worker`.
```
Comment (m-rousse) [inline content/en/synthetics/private_locations/configuration.md:127]:
```suggestion
Example for Docker: `docker run --env SECRET=value datadog/synthetics-private-location-worker --environmentVariableOverride`.
```

Comment (HugoPct) [reply content/en/synthetics/private_locations/configuration.md:127]: btw should it be --env SECRET=value or --env SECRET ?

Comment (m-rousse) [reply content/en/synthetics/private_locations/configuration.md:127]: Both work:

- `--env SECRET` will expose `SECRET=<value from the host>` in the container
- `--env SECRET=value` will expose `SECRET=value` in the container

Comment (HugoPct) [reply content/en/synthetics/private_locations/configuration.md:127]: so option 2 = using `variableOverride`

File: content/en/synthetics/private_locations/configuration.md [hunk 2]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
...
+
+`variableOverride`
+: **Type**: String <br>
+Overrides the variables used in tests running on the Private Location. Format: `SECRET_VAR=value`.
+All variables imported this way will be obfuscated.
+
+`environmentVariableOverride`
+: **Type** String <br>
+Allows the usage of environment variables in the Private Location. It requires the environment variables to be imported in the containerized environment.
+Example for Docker: `docker run --env SECRET=value gcr.io/datadoghq/synthetics-private-location-worker --environmentVariableOverride SECRET`.
```
Comment (HugoPct) [inline content/en/synthetics/private_locations/configuration.md:127]:
```suggestion
Example for Docker: `docker run --env VARIABLE gcr.io/datadoghq/synthetics-private-location-worker --environmentVariableOverride SECRET`.
```

File: content/en/synthetics/private_locations/configuration.md [hunk 3]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
 `maxNbRedirects`
 : **Type**: Number <br>
 **Default**: `20`<br>
-Maximum number of redirections for HTTP and WebSocket tests. 
+Maximum number of redirections for HTTP and WebSocket tests.
+
+`variableOverride`
+: **Type**: String <br>
+Overrides the variables used in tests running on the Private Location. Format: `SECRET_VAR=value`.
```
Comment (HugoPct) [inline content/en/synthetics/private_locations/configuration.md:121]:
```suggestion
Overrides the variables used in tests running on the Private Location. Format: `LOCAL_VAR=value`.
```

File: content/en/synthetics/private_locations/configuration.md [hunk 4]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
...
+Maximum number of redirections for HTTP and WebSocket tests.
+
+`variableOverride`
+: **Type**: String <br>
+Overrides the variables used in tests running on the Private Location. Format: `VARIABLE=value`.
+All variables imported this way will be obfuscated.
+
+`environmentVariableOverride`
+: **Type** String <br>
+Allows the usage of environment variables in the Private Location. It requires the environment variables to be imported in the containerized environment.
```
Comment (HugoPct) [inline content/en/synthetics/private_locations/configuration.md:126]:
```suggestion
Overrides variables used in tests running on the Private Location with environment variables. It requires the environment variables to be imported in the containerized environment.
```

File: content/en/synthetics/private_locations/configuration.md (hunk 1 above)
Comment (m-rousse) [reply content/en/synthetics/private_locations/configuration.md]: Yes

**Review by m-rousse — APPROVED** [review]

File: content/en/synthetics/private_locations/configuration.md [hunk 5]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
 `maxNbRedirects`
 : **Type**: Number <br>
 **Default**: `20`<br>
-Maximum number of redirections for HTTP and WebSocket tests. 
+Maximum number of redirections for HTTP and WebSocket tests.
+
+`variableOverride`
+: **Type**: String <br>
+Overrides the variables used in tests running on the Private Location. Format: `VARIABLE=value`.
+All variables imported this way will be obfuscated.
```
Comment (cswatt) [inline content/en/synthetics/private_locations/configuration.md:122]:
```suggestion
All variables imported this way are obfuscated.
```

File: content/en/synthetics/private_locations/configuration.md [hunk 6]
```diff
@@ -114,7 +114,18 @@ Overrides the port for the private location status probes.
...
+
+`environmentVariableOverride`
+: **Type** String <br>
+Overrides variables used in tests running on the Private Location with environment variables. It requires the environment variables to be imported in the containerized environment.
+Example for Docker: `docker run --env VARIABLE gcr.io/datadoghq/synthetics-private-location-worker --environmentVariableOverride VARIABLE`.
+All variables imported this way will be obfuscated.
```
Comment (cswatt) [inline content/en/synthetics/private_locations/configuration.md:128]:
```suggestion
All variables imported this way are obfuscated. 
```

**Review by cswatt — CHANGES_REQUESTED** [review]

**Review by cswatt — APPROVED** [review]

**Pull request merged**